### PR TITLE
Record the rustc target platform into the SBOM

### LINKED ISSUE #528: Record the target platform in the SBOM
The target for which the SBOM is generated should be recorded in it.

It would be nice to also append it to the end of the SBOM filename, to make it very clear that the SBOM is generated for a specific platform (with an option to disable this).

### PR DESCRIPTION
**Pull request by Shnatsel**
Fixes #528

### CODE REVIEW

File: cargo-cyclonedx/src/generator.rs [hunk 1]
```diff
@@ -311,6 +316,15 @@ impl SbomGenerator {
 
         metadata.tools = Some(Tools(vec![tool]));
 
+        let properties = match self.config.target.as_ref().unwrap() {
+            SingleTarget(target) => vec![Property::new("rustcTarget", &target)],
+            AllTargets => all_known_targets()
+                .into_iter()
+                .map(|target| Property::new("rustcTarget", &target))
```
Comment (lfrancke) [inline cargo-cyclonedx/src/generator.rs:323]: I like the idea in general. Have you checked if there are any plans/standards/best practices for what this property could be called? Other tools must have the same idea/concept in general (e.g. Go/C/C++)

**Review by lfrancke — COMMENTED** [review]
(didn't look at the code yet, just one question)

File: cargo-cyclonedx/src/generator.rs (hunk 1 above)
Comment (Shnatsel) [reply cargo-cyclonedx/src/generator.rs:323]: I have not really checked for other uses of `properties`, just skimmed the standardized fields of `metadata`. 

This encodes Rust target triples specifically, e.g. `x86_64-unknown-linux-gnu`, so this is not going to be interoperable with anything else. Hence the use of `properties` instead.

File: cargo-cyclonedx/src/generator.rs [hunk 2]
```diff
@@ -311,6 +316,15 @@ impl SbomGenerator {
 
         metadata.tools = Some(Tools(vec![tool]));
 
+        let properties = match self.config.target.as_ref().unwrap() {
+            SingleTarget(target) => vec![Property::new("rustcTarget", &target)],
```
Comment (lfrancke) [inline cargo-cyclonedx/src/generator.rs:320]:
```suggestion
            SingleTarget(target) => vec![Property::new("rustcTarget", target)],
```

Comment (lfrancke) [reply cargo-cyclonedx/src/generator.rs:320]: clippy lint

File: cargo-cyclonedx/src/generator.rs (hunk 1 above)
Comment (lfrancke) [reply cargo-cyclonedx/src/generator.rs:323]: Understood. In that case, does it make sense to use a namespaced property instead?

https://github.com/CycloneDX/cyclonedx-property-taxonomy
https://github.com/CycloneDX/cyclonedx-property-taxonomy/blob/main/cdx.md

**Review by lfrancke — COMMENTED** [review]
Tried, tested, works etc.
Happy to approve in principle but I think I'd like to talk about the name of the property a bit first.
This is the first PR (I believe) to introduce a custom property so it'll be a bit more involved. The next one will be easier.

File: cargo-cyclonedx/src/generator.rs (hunk 1 above)
Comment (lfrancke) [reply cargo-cyclonedx/src/generator.rs:323]: Found the property for Go: https://github.com/CycloneDX/cyclonedx-property-taxonomy/blob/main/cdx/gomod.md

Comment (Shnatsel) [reply cargo-cyclonedx/src/generator.rs:323]: Yes, it does. Good point.

**Comment by lfrancke** [comment]
I think this is basically good to go, we just need to wait for the result of https://github.com/CycloneDX/cyclonedx-property-taxonomy/issues/75

**Comment by Shnatsel** [comment]
That property needs a fair bit of design work. We need to specify where it is legal for it to appear: on metadata only, or does it also appear on a `component` to specify what target is it for? When recording dependencies for all platforms, do we have a boolean flag to indicate it, or do we list all platforms considered? If the latter, does it make sense to make that property an array or a set, instead of specifying it multiple times?

**Comment by Shnatsel** [comment]
I've opened https://github.com/CycloneDX/cyclonedx-property-taxonomy/pull/78/ upstream. Once that's merged, this PR will need to be reworked to match that schema.

**Comment by Shnatsel** [comment]
Superseded by #762

**Pull request closed without merging**